scripts/leetcode_directory_md.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python3
+
+from urllib.request import Request, urlopen
+from pathlib import Path
+from os import listdir
+import json
+from dataclasses import dataclass
+
+
+@dataclass
+class Task:
+    id: str
+    title: str
+    solution: str
+    difficulty: str
+
+def fetch_leetcode_folder_tasks(solutions_folder: Path) -> list[Task]:
+    # Fetch tasks info from the leetcode API.
+    req = Request(
+        url="https://leetcode.com/api/problems/algorithms/", 
+        headers={"User-Agent": "Mozilla/5.0 (X11; U; Linux i686) Gecko/20071127 Firefox/2.0.0.11"}
+    )
+
+    content = urlopen(req).read()
+    content_dict = json.loads(content)

```

Comment:
Using the `requests`, changes this to `resp.json()`

```suggestion
    content_dict = resp.json()
```